src/cd-profile-db.c

```diff
@@ -69,6 +69,7 @@ cd_profile_db_load (CdProfileDb *pdb,
 			     CD_CLIENT_ERROR_INTERNAL,
 			     "Can't open database: %s\n",
 			     sqlite3_errmsg (priv->db));
+		sqlite3_free (error_msg);
```

Comment:
I think the cleanest would probably be defining a custom type alias to be autocleaned like in this example:

https://github.com/GNOME/libshumate/blob/fafc60331b6001781c9ed4824ad47141e0ae57fd/shumate/shumate-file-cache.c#L79-L80

https://github.com/GNOME/libshumate/blob/fafc60331b6001781c9ed4824ad47141e0ae57fd/shumate/shumate-file-cache.c#L607

Reply:
Oh, and just so that it is not too clean, we will need explicit `sqlite3_free` at least before the following two `exec`s:

https://github.com/hughsie/colord/blob/d8076a16ea1f4437cca3799d414faf40cddae955/src/cd-mapping-db.c#L180-L182
https://github.com/hughsie/colord/blob/d8076a16ea1f4437cca3799d414faf40cddae955/src/cd-mapping-db.c#L193-L199

This is because they re-use the variable but the autocleanup handlers only run when the scope containing the variable is left, not when something re-assigns it (unlike [smart pointers in C++](https://stackoverflow.com/questions/27302548/reassigning-smart-pointers#27302643), which override `operator=`, or anything in Rust, which calls “destructor” [when value is no longer needed](https://doc.rust-lang.org/std/ops/trait.Drop.html)).